MarquezProject/marquez
# GitHub Action workflow to validate changes to Helm chart

### Problem

When changes are made to the Marquez Helm chart, they need to be manually verified by installing and testing the chart into a local instance of Kubernetes (ex: Minikube). This process is both time consuming and prone to error.

### Solution

An automated GitHub Actions workflow has been created that performs the following checks whenever a pull request is opened that affects the Helm chart.

- Lint the chart to verify it is valid to the Helm specification
- Create a [kind](https://kind.sigs.k8s.io/) Kubernetes cluster and use it to deploy the Helm chart
- Execute the Helm chart tests, which verify the deployment completed successfully and the pods/services are available

All of these checks can executed locally as well, using the [act](https://github.com/nektos/act) utility. Additional instructions have been added to the contribution guide.

### Checklist

- [X] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [X] Your changes are accompanied by tests (_if relevant_)
- [X] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [X] You've updated any relevant documentation (_if relevant_)
- [X] You've updated the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
